test/it/EditVatLinkAcceptanceSpec.scala

```diff
@@ -74,4 +104,110 @@ class EditVatLinkAcceptanceSpec extends ServerAcceptanceSpec with WireMockSbrCon
       response.status shouldBe CREATED
     }
   }
+
+  feature("edit a non-existing VAT records parent link") {
+    scenario("by VAT reference (vatref) and UBRN for a specific period") { wsClient =>
+      Given(s"a VAT record with $TargetVAT for $TargetPeriod does not exist")
+      stubSbrControlApiFor(aVatParentLinkEditRequest(withVatRef = TargetVAT, withPeriod = TargetPeriod)
+        .withHeader(HeaderNames.CONTENT_TYPE, equalTo(PatchJson))
+        .withRequestBody(equalToJson(VATEditParentLinkPatchBody))
+        .willReturn(aNotFoundResponse()))
+
+      When(s"an edit request for the VAT unit with $TargetVAT is requested for $TargetPeriod")
+      val response = await(wsClient
+        .url(s"/v1/periods/${Period.asString(TargetPeriod)}/edit/vats/${TargetVAT.value}")
+        .withHeaders((HeaderNames.CONTENT_TYPE, JSON))
+        .post(VATEditParentLinkPostBody))
+
+      Then(s"the status of the edit of the parent LEU from $ParentLEU to $NewParentLEU will be sent back")
```

Comment:
Although we write these in ScalaTest rather than Cucumber, the reason we use this BDD style is to focus on the business feature rather than the implementation detail that unit tests look at.

If I was to delete the code steps and give the product owner the feature / scenario / Given / When / Then lines - would they understand the intent of the test?

The "Then" should probably just be something like "a Not Found response is returned".

Reply:
That sounds like a more informative approach!